### PR TITLE
Feature/answer author indicator improve myself

### LINKED ISSUE #1314: Answer author indicator: improve for "myself"
## Overview
Screenshot 2023-03-01 at 16 40 16

When we load our own answers as read-only, give a way to load them as current.

Also improve a bit the design.

## Acceptance criteria
- [ ] Move the score in front of the text as a score ring (of the size of the text)
- [ ] When the user is yourself (pass the info from item-by-id), use "Submitted by yourself on ..." instead of the login. (no link needed)
- [ ] When the user is yourself AND not observing, add a link (a button if it can be small enough) "Edit it" which links to the same answer (by id) with "loadAsCurrent" parameter.
- [ ] When the user is yourself AND not observing, add a link (a button if it can be small enough) "Back to your current answer" which links to the same page without any "answer" arg.

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1314

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

...

## Test cases

- [ ] Case 1:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/1293400563985240032;p=7523720120450464843;a=0;answerId=5264059411445262334)
  3. Then I see answer indicator with buttons for current user
  4. And I click on "History" tab
  5. And I see the paddings are correct

- [ ] Case 2:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/1293400563985240032;p=7523720120450464843;a=0;answerId=7669589354524085472)
  3. Then I see answer indicator without buttons for non current user
  4. And I click on "History" tab
  5. And I see the paddings are correct

- [ ] Case 3:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/6379723280369399253;p=7523720120450464843;a=0;answerId=5264059411445262334)
  3. Then I see full size task and answer indicator with buttons for current user
  4. And I click on "History" tab
  5. And I see the paddings are correct

- [ ] Case 4:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/6379723280369399253;p=7523720120450464843;a=0;answerId=3193667051655421989)
  3. Then I see  full size task and answer indicator without buttons for non current user
  4. And I click on "History" tab
  5. And I see the paddings are correct

- [ ] Case 5:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/6379723280369399253;p=7523720120450464843;a=0?watchedGroupId=117532200658920233&watchUser=0)
  3. Then I see no answer indicator

- [ ] Case 6:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/7523720120450464843;p=;a=0)
  3. Then I see the paddings are correct

- [ ] Case 7:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/1293400563985240032;p=7523720120450464843;a=0;answerId=5264059411445262334)
  3. Then I see answer indicator for current user
  4. And I click on "Edit it" button
  5. And I see opened task

- [ ] Case 8:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/1293400563985240032;p=7523720120450464843;a=0;answerId=5264059411445262334)
  3. Then I see answer indicator for current user
  4. And I click on "Back to your current answer" button
  5. And I see opened task

- [ ] Case 9:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/answer-author-indicator-improve-myself/en/a/6379723280369399253;p=;pa=0;answerId=2656751988652039073)
  3. Then I see answer indicator for current user with score ring

### CODE REVIEW

**Review by smadbe — COMMENTED** [review]
https://github.com/France-ioi/AlgoreaFrontend/assets/1053150/12b57633-c49a-40e1-8ae2-cc0041e5d549

There is a transition from:
- fat bar while loading: Screenshot 2023-11-13 at 10 05 04
- answer bar with top padding:Screenshot 2023-11-13 at 10 05 14
- final position  Screenshot 2023-11-13 at 10 05 20

Is there something we can do about it easily ? (the fat bar thing is not really visible but the jump after the task loading is)

File: src/app/items/item-by-id.component.html [hunk 1]
```diff
@@ -1,76 +1,92 @@
...
-        *ngrxLet="currentTab$ as currentTab"
-      >
-        <ng-container *ngIf="currentTab?.isTaskTab || (itemContentComponent?.isTaskLoaded$ | async) || currentTab?.tag === 'alg-content' || currentTab?.tag === 'alg-children-edit'">
-          <alg-error *ngIf="answerLoadingError$ | async as error; else noAnswerLoadingError">
-            <ng-container *ngIf="!error.fallbackLink; else fallback" i18n>Unable to load the answer</ng-container>
-            <ng-template #fallback>
-              <ng-container i18n>
-                Unable to load the answer, <a [routerLink]="error.fallbackLink" class="alg-link">back to the regular task page</a>.
+    <ng-container *ngrxLet="isWatching$ as isWatching">
+      <ng-container *ngrxLet="fullFrameContentDisplayed$; let fullFrameContentDisplayed">
```
Comment (smadbe) [inline src/app/items/item-by-id.component.html:4]: Be consistent, either `x$ as x` or `x$; let x`.

To match their doc, it seems better to use `x$ as x`

File: src/app/services/layout.service.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export class LayoutService implements OnDestroy {
     distinctUntilChanged(),
   );
   fullFrameContentDisplayed$ = this.contentDisplayType$.pipe(map(t => t === ContentDisplayType.ShowFullFrame));
+  fullFrameWithLeftPaddingContentDisplayed$ = this.contentDisplayType$.pipe(map(t => t === ContentDisplayType.Show));
```
[the remark this answers is not in the document]
Comment (smadbe) [reply src/app/services/layout.service.ts:35]: Probably, `Show` should be renamed `ShowWithLeftPadding`. Is it what it is in practice ?

And I still don't get your `fullFrameWithLeftPaddingContentDisplayed`which just matches it is a task... why "full frame" ? The existing (previous) "full frame" defines tasks which announce they have to be displayed using the full frame.

File: src/app/items/item-by-id.component.html
Comment (smadbe) [inline src/app/items/item-by-id.component.html:1]: This file change is not really reviewable the way GH displays it to me :-/  I cannot see all changes.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/services/layout.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/services/layout.service.ts:35]: I've supposed it requires more generic name

File: src/app/app.component.html [hunk 3]
```diff
@@ -2,27 +2,32 @@
...
-        (elLoaded)="onLoaded($event)"
-      >
-        <div class="main-content" id="main-container">
-          <div class="main-content-bottom-padding alg-flex-1">
-            <router-outlet></router-outlet>
+  <ng-container *ngrxLet="{ fullFrameContentDisplayed: fullFrameContentDisplayed$, fullFrameWithLeftPaddingContentDisplayed: fullFrameWithLeftPaddingContentDisplayed$ } as layout">
+    <ng-container *ngrxLet="isWatching$ as isWatching">
+      <div class="right-container" [ngClass]="{ collapsed: !leftMenu.shown, animated: leftMenu.animated, 'thread-collapsed': !(isDiscussionVisible$ | async) }">
+        <alg-top-bar
+          [showBreadcrumbs]="!scrolled && !layout.fullFrameWithLeftPaddingContentDisplayed && !(isNarrowScreen$ | async) && !!(canShowBreadcrumbs$ | async)"
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/app.component.html]: Yep, thank you. Fixed

File: src/app/items/item-by-id.component.html (hunk 1 above)
Comment (Iloveall) [reply src/app/items/item-by-id.component.html]: Fixed

**Comment by Iloveall** [comment]
> Screen.Recording.2023-11-13.at.10.03.00.mov 
> There is a transition from:
> 
> * fat bar while loading: Screenshot 2023-11-13 at 10 05 04
> * answer bar with top padding:Screenshot 2023-11-13 at 10 05 14
> * final position  Screenshot 2023-11-13 at 10 05 20
> 
> Is there something we can do about it easily ? (the fat bar thing is not really visible but the jump after the task loading is)

I've add easiest solution for my point

File: src/app/items/item-by-id.component.html [hunk 4]
```diff
@@ -1,76 +1,92 @@
...
+                  class="indicator"
+                  [ngClass]="{ 'no-left-padding': !fullFrameContentDisplayed }"
+                >
+                  <ng-container *ngIf="userProfile$ | async as userProfile">
+                    <alg-answer-author-indicator
+                      *ngIf="!!currentTab?.isTaskTab && taskConfig.initialAnswer"
+                      [answer]="taskConfig.initialAnswer"
+                      [userProfile]="userProfile"
+                      [itemData]="itemData"
+                      [isWatching]="isWatching"
```
Comment (smadbe) [inline src/app/items/item-by-id.component.html:57]: same with `isWatching` does it make sense to send it while the non-dump component could register to it?

Comment (Iloveall) [reply src/app/items/item-by-id.component.html]: Fixed

**Review by smadbe — APPROVED** [review]

**Pull request merged**